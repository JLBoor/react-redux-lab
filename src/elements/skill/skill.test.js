import React from 'react';
import ReactDOM from 'react-dom';
import SkillComponent from './skill.component';
import renderer from 'react-test-renderer';


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SkillComponent />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should render a green label for the web category', () => {
    const component = renderer.create(<SkillComponent category="web"></SkillComponent>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render an orange label for the microservices category', () => {
    const component = renderer.create(<SkillComponent category="microservices"></SkillComponent>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render a red label for the devops category', () => {
    const component = renderer.create(<SkillComponent category="devops"></SkillComponent>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it('should render a navy label for the bigdata category', () => {
    const component = renderer.create(<SkillComponent category="bigdata"></SkillComponent>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
