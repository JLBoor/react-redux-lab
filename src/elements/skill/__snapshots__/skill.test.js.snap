// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a green label for the web category 1`] = `
<span
  className="uk-label uk-margin-right"
  style={
    Object {
      "backgroundColor": "#29BF12",
    }
  }
/>
`;

exports[`should render a navy label for the bigdata category 1`] = `
<span
  className="uk-label uk-margin-right"
  style={
    Object {
      "backgroundColor": "#2F4858",
    }
  }
/>
`;

exports[`should render a red label for the devops category 1`] = `
<span
  className="uk-label uk-margin-right"
  style={
    Object {
      "backgroundColor": "#D63230",
    }
  }
/>
`;

exports[`should render an orange label for the microservices category 1`] = `
<span
  className="uk-label uk-margin-right"
  style={
    Object {
      "backgroundColor": "#FF9914",
    }
  }
/>
`;
